setup.py

```diff
@@ -37,11 +37,14 @@ class Components:
         .decode()
     )
 
-    __version__ = v[: v.rfind("-")].replace("-", ".") if "-" in v else v
+    # make git describe output PEP440 compliant
+    __version__ = v.replace("-", ".", 1).replace("-", "+", 1)
 
     # if version is not a valid PEP440 version, then create a bogus version
     # that will be used only for development purposes which appends the commit hash
-    if not re.match(r"^\d+(\.\d+)*$", __version__):
+    try:
+        version_obj = parse_version(__version__)
+    except ValueError:
         __version__ = "0.0.dev0+g" + (
             subprocess.run(
                 ["git", "rev-parse", "--short", "HEAD"], stdout=subprocess.PIPE
```

Comment:
I will tag @ferdonline for a quick feedback here.

I haven't checked PEP440 and our version regex but wondering if there is better approach than automatically setting version like `0.0xx` in except block (to avoid having botched release versions accidentally ).

Reply:
there is a (recommended) approach , `setuptools_scm` with `use_scm_version`, that will automatically handle everything. but that would disallow the possibility of setting the versions ourselves in:
```python
# allow to override version during development/testing
    if "NEURON_WHEEL_VERSION" in os.environ:
        __version__ = os.environ["NEURON_WHEEL_VERSION"]
```

Our release policy sp

Reply:
> setting version like `0.0xx` in except block

since we (manually) rely on `git describe` and shallow clones ...

Reply:
> there is a (recommended) approach , setuptools_scm with use_scm_version, that will automatically handle everything

As discussed, we can go with this then as a separate PR. (and we can merge this to unblock failing nightly CIs)